### ISSUE TITLE
change_map command does not work

### ISSUE THREAD

**Issue by themisak** [teeworlds/teeworlds#1841]
Hey,

today I tried to change the map on my server with change_map ctf2 in rcon and it did not work. No reaction by the server. Only sv_map worked. 

Did I do anything wrong? Whats actually the difference between sv_map and change_map ?

Cheers!

**Comment by sepulzera** [comment]
sv_map sets the default map to use on the server.
change_map should ... change the map, aside of sv_map or sv_maprotation.

**Comment by sepulzera** [comment]
Okay here we go:
change_map will queue the given map as "next map". Additionally, it tries to end the current match to rotate to the next map.
I suspect your match didn't start yet, as you were still in warm up. Then it won't work.

**Comment by Dune-jr** [comment]
There still seems to be a problem here. I think the current (intended) behaviors are:

- `sv_map X`: load map X (used to not take effect immediately, now does)
- `change_map X`: end the game (prints the scoreboard), then load map X

The different is small enough that I think `change_map` should probably be outright removed.

**Comment by sepulzera** [comment]
change_map will copy the requested map to sv_map, see https://github.com/teeworlds/teeworlds/blob/19e918d0f15011c6231d05e5cbe491516de2deeb/src/game/server/gamecontroller.cpp#L891-L898
And sv_map will change the map, if the current map does not match, see https://github.com/teeworlds/teeworlds/blob/19e918d0f15011c6231d05e5cbe491516de2deeb/src/engine/server/server.cpp#L1293-L1299


> The different is small enough that I think change_map should probably be outright removed.

I am not quite sure if that would yield negative sideeffects to the maprotation etc.

**Comment by sepulzera** [comment]
We could also go for a different approach: Don't make change_map force end the current match. Then you can change the map for the next match without interrupting the current one.
Though this is what the maprotation is for, change_map could be more simple to use without overriding any existing maprotation.

**Comment by Dune-jr** [comment]
That would make the difference between `sv_map` and `change_map` even smaller. I think it's just confusing.

**Comment by oy** [comment]
Calling change_map with a parameter switches to the next map in the map rotation.

**Comment by themisak** [comment]
> Calling change_map with a parameter switches to the next map in the map rotation.

So you if the current map is ctf2 and i put in r2 "change_map ctf3" it will change the map to ctf3 after the match has finished?

So If I want to change the map in the middle of the rould I should use sv_map?

**Comment by oy** [comment]
No, it shows the scoreboard, to notify everyone that the match is over and then it changes to ctf3.

**Comment by oy** [comment]
Yeah, sry typo.
Calling change_map without a parameter switches to the next map in the map rotation.

**Comment by themisak** [comment]
So with no map rotation in place change_map doesent do anything? and it is only to be used without a parameter?

Because in 0.6.x I could type change_map ctf2 and it would change map to ctf2. That doesent work in 0.7. anymore so change_map has a different function now. Is that correct ?

**Comment by sepulzera** [comment]
@themisak Please read my second comment, again.

**Comment by Dune-jr** [comment]
> Because in 0.6.x I could type change_map ctf2 and it would change map to ctf2. That doesent work in 0.7. anymore so change_map has a different function now. Is that correct?

It should still work in 0.7. If it doesn't, I smell a bug.

**Comment by themisak** [comment]
Please try it Dune. On my Server it doesent work. You play it please and say if it works

**Comment by Dune-jr** [comment]
I will, @themisak

**Comment by sepulzera** [comment]
> I think it only does not work during warum up time when I am alone on a server. Then only sv_map works

@themisak Seriously, that is exactly what I told you with my second comment! >_<

**Comment by sepulzera** [comment]
I will take a closer look on it.

**Comment by nheir** [comment]
https://github.com/teeworlds/teeworlds/blob/d1bf48f42cbf512428c311bb7c06b7ed7be5b034/src/game/server/gamecontroller.cpp#L607-L615
